pkg/controller/managedcluster/managed_cluster_controller.go

```diff
@@ -129,75 +136,61 @@ func clusterClaimCluster(meta metav1.Object) bool {
-				ApplicationManagerConfig:   kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
-				CertPolicyControllerConfig: kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
-				IAMPolicyControllerConfig:  kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
-				PolicyController:           kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
-				SearchCollectorConfig:      kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
-			},
-		}
-	case clusterType == clusterTypeHypershift:
-		return &kacv1.KlusterletAddonConfig{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: name,
-				Name:      name,
-			},
-			Spec: kacv1.KlusterletAddonConfigSpec{
-				ClusterName:                name,
-				ClusterNamespace:           name,
-				ApplicationManagerConfig:   kacv1.KlusterletAddonAgentConfigSpec{Enabled: false},
-				CertPolicyControllerConfig: kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
-				IAMPolicyControllerConfig:  kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
-				PolicyController:           kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
-				SearchCollectorConfig:      kacv1.KlusterletAddonAgentConfigSpec{Enabled: false},
-			},
-		}
-	default:
-		return nil
+func clusterClaimKAC(clusterName string) *kacv1.KlusterletAddonConfig {
+	return &kacv1.KlusterletAddonConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: clusterName,
+			Name:      clusterName,
+		},
+		Spec: kacv1.KlusterletAddonConfigSpec{
+			ClusterName:                clusterName,
+			ClusterNamespace:           clusterName,
+			ClusterLabels:              map[string]string{"vendor": "OpenShift"}, // Required for object to be created
+			ApplicationManagerConfig:   kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
+			CertPolicyControllerConfig: kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
+			IAMPolicyControllerConfig:  kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
+			PolicyController:           kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
+			SearchCollectorConfig:      kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
+		},
 	}
 }
 
-const (
-	clusterTypeHypershift   string = "Hypershift"
-	clusterTypeClusterClaim string = "ClusterClaim"
-)
+func defaultKAC(clusterName string) *kacv1.KlusterletAddonConfig {
+	return &kacv1.KlusterletAddonConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: clusterName,
+			Name:      clusterName,
+		},
+		Spec: kacv1.KlusterletAddonConfigSpec{
+			ClusterName:                clusterName,
+			ClusterNamespace:           clusterName,
+			ApplicationManagerConfig:   kacv1.KlusterletAddonAgentConfigSpec{Enabled: false},
+			CertPolicyControllerConfig: kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
+			IAMPolicyControllerConfig:  kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
+			PolicyController:           kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
+			SearchCollectorConfig:      kacv1.KlusterletAddonAgentConfigSpec{Enabled: false},
```

Comment:
I think we need to enable/disable addon via checking the overrides in mch .
```
overrides:
    components:
    - enabled: true
      name: console
    - enabled: true
      name: insights
    - enabled: true
      name: grc
    - enabled: true
      name: cluster-lifecycle
    - enabled: true
      name: volsync
    - enabled: true
      name: multicluster-engine
    - enabled: true
      name: search
    - enabled: true
      name: app-lifecycle
    - enabled: false
      name: cluster-backup
```

Reply:
following up : add a TODO here,  we can use cma to check addon controller is installed or not here.

Reply:
Good point, this seems worth to make another PR to fix since it affect addon installation in "hypershift", "clusterclaim" as well.